File: TypeScript/Newton'sMethod.ts
import * as readline from 'readline'

const readInterface = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})


const NewtonsMethod = (value: number) =>{
    
    if(value < 0){
        throw new Error('1以上の整数を入力してください')
    }

    const MAXLOOP = 100
    const EV = 0.0000001
    
    let x0 = value
    let x1 = 0

    let delta = EV

    for(let i = 0; i < MAXLOOP && Math.abs(delta) >= EV; i++){

        delta = (x0 * x0 - value)/(2.0 * x0)
        x1 = x0 - delta
        x0 = x1
    }

    return x1
}

readInterface.question('平方根を求める数値を入力してください',(n:string)=>{
    readInterface.close()

    const value = Number(n)

    try{
        const result = NewtonsMethod(value)
        console.log(`${value}の平方根(近似値)は${result}です。`)
    }
    catch(e){
        console.log(e)
    }
})

File: Python/newton's_method.py
import math


def newtons_method(value: float)->float:
    """
    与えられた整数の平方根をニュートン法により求める

    Parameters
    ----------
    value: float
        対象整数

    Returns
    -------
    result: float
        valueの平方根
    """
    if type(value) is not float:
        raise TypeError('入力された整数がfloat型ではありません')
    elif value < 0:
        raise ValueError('1以上の整数を入力してください')


    MAXLOOP = 100
    EV = 0.0000001
    x0 = value
    x1= 0

    delta = EV

    for i in range(MAXLOOP):
        if math.fabs(delta) < EV:
            break

        delta = (x0 * x0 - value)/(2.0 * x0)
        x1 = x0 - delta
        x0 = x1

    return x1

if __name__ == '__main__':

    try:
        value = float(input('平方根を求める数値を入力してください'))
        # value = 'a'

        result = newtons_method(value)
        
        print('{}の平方根（近似値）は{:.08f}です。'.format(value, result))
        
    except TypeError as e:
        print(e)
    except ValueError as e:
        print(e)

